apps/crn-server/src/handlers/external-author/index-handler.ts

```diff
@@ -15,45 +19,58 @@ import {
   algoliaIndex,
   appName,
   baseUrl,
+  contentfulAccessToken,
+  contentfulEnvId,
+  contentfulManagementAccessToken,
+  contentfulSpaceId,
+  isContentfulEnabledV2,
 } from '../../config';
-import {
-  ExternalAuthorDataProvider,
-  ExternalAuthorSquidexDataProvider,
-} from '../../data-providers/external-authors.data-provider';
+import ExternalAuthors, {
+  ExternalAuthorsController,
+} from '../../controllers/external-authors';
+import { ExternalAuthorContentfulDataProvider } from '../../data-providers/contentful/external-authors.data-provider';
+import { ExternalAuthorSquidexDataProvider } from '../../data-providers/external-authors.data-provider';
 import { getAuthToken } from '../../utils/auth';
 import logger from '../../utils/logger';
 import { sentryWrapper } from '../../utils/sentry-wrapper';
 import { ExternalAuthorEvent, ExternalAuthorPayload } from '../event-bus';
 
 export const indexExternalAuthorHandler =
   (
-    externalAuthorDataProvider: ExternalAuthorDataProvider,
+    externalAuthorController: ExternalAuthorsController,
     algoliaClient: AlgoliaSearchClient,
   ): EventBridgeHandler<ExternalAuthorEvent, ExternalAuthorPayload> =>
   async (event) => {
     logger.debug(`Event ${event['detail-type']}`);
 
-    try {
-      const externalAuthor = await externalAuthorDataProvider.fetchById(
-        event.detail.payload.id,
-      );
+    const getExternalAuthor = async () => {
+      try {
+        const externalAuthor = await externalAuthorController.fetchById(
+          event.detail.payload.id,
+        );
 
-      if (!externalAuthor) {
+        logger.debug(`Fetched external author ${externalAuthor.displayName}`);
+        return externalAuthor;
+      } catch {
         await algoliaClient.remove(event.detail.payload.id);
-        return;
+        return null;
       }
+    };
 
-      logger.debug(`Fetched external author ${externalAuthor.displayName}`);
```

Comment:
I've made this piece

```
const externalAuthor = await externalAuthorDataProvider.fetchById(
        event.detail.payload.id,
      );
```

to be wrapped by a try catch block because now that we are using the controller, if the External Author is not found is going to throw a not found error, it's not going to return null

Reply:
thats correct, yeah